src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -570,11 +563,34 @@ private IAccount GetAccountByUserFlow(IEnumerable<IAccount> accounts, string use
             foreach (var account in accounts)
             {
                 string accountIdentifier = account.HomeAccountId.ObjectId.Split('.')[0];
-                if (accountIdentifier.EndsWith(userFlow.ToLower()))
+                if (accountIdentifier.EndsWith(userFlow.ToLower(CultureInfo.CurrentCulture), StringComparison.InvariantCulture))
                     return account;
             }
 
             return null;
         }
+
+        internal /*for test only*/ string CreateRedirectUri()
+        {
+            if (Uri.TryCreate(_microsoftIdentityOptions.RedirectUri, UriKind.Absolute, out Uri uri))
+            {
+                if (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps)
+                {
+                    return _microsoftIdentityOptions.RedirectUri;
+                }
+                else
+                {
+                    _logger.LogInformation("MicrosoftIdentityOptions RedirectUri value must have a Uri Scheme " +
+                        "of http or https in order to be a valid RedirectUri. ");
+                }
+            }
+
+            var request = CurrentHttpContext.Request;
+            return UriHelper.BuildAbsolute(
+                request.Scheme,
```

Comment:
I don't really understand the value here, but I think you've discussed this before.

Reply:
Imagine that you run locally on http;//localhost:1234
then you deploy to azure (bg.azurewebapps.net) 

if you don't use redirectUri, you don't need to change your config file.
Now if you add a reverse proxy that tweaks your URI, you might want to specify it (there are cases where that's needed. See [Bug] Redirect URI is set to http instead of https when deploying to Azure App Service for Docker container (Linux) #115

Reply:
what JM said. :)

---
In reply to: [420358960](https://github.com/AzureAD/microsoft-identity-web/pull/144#discussion_r420358960) [](ancestors = 420358960)

Reply:
Thx for the explanation!